db/sqlinterfaces.c

```diff
@@ -3884,6 +3884,16 @@ static void clear_parameters_to_bind(struct sql_state *rec)
     }
 }
 
+static int dont_cache_sql(struct sqlclntstate *clnt, const char *sql)
+{
+    //if (clnt->ddlflag) return 1;
+    if (strncmp(sql, "analyze", 7) == 0) return 1;
```

Comment:
May want to do strncasecmp, and skip any leading spaces.

Reply:
I will check for case. Will consume whitespace when we initially set clnt->sql in a separate PR.
Testcase with uppercase ANALYZE you mean?

Reply:
I meant a test that the cache is properly cleared on user change.